changelog.md

```diff
@@ -1,14 +1,41 @@
 ## Version 1.19.0-dev
  ### Add-ons client
+   * Show the server ID in the Add-ons Manager
  ### Add-ons server
  ### Campaigns
    * Descent into Darkness
      * S3: player now no longer gains any exp upon killing rats and crawlers, buffed puzzle exp to compensate
+   * Eastern Invasion
+     * Prevent Dacyn from picking up a different staff (issue #8885)
+     * Reduce Barrow/Pyre Wight vulnerability to arcane
+     * S14: indicate uncleared swamp hexes with "swamp reed" images
+     * Many minor bugfixes
+   * Liberty
+     * Custom unit and portrait for Lord Maddock
+   * Sceptre of Fire
+     * S7: fewer and slower enemies, to balance the buff to the Elvish Outrider unit type
+   * Tutorial
+     * Better timing for some of the hints
+   * Under the Burnings Suns
+     * S01: the luck involved in getting units from villages is now fairer
+   * Winds of Fate
+     * Avoid a Lua error that only triggered on Windows in locales that use commas as the decimal separator
+     * Rebalanced enemies and enemy gold from S6 onwards
+     * S08: clarify the objectives
  ### Editor
+   * Allow loading .mask files
  ### Multiplayer
+   * The default settings for the timer are now 240 seconds turn bonus, with no action bonus
+   * The full map is no longer revealed in linger mode by default, so it behaves similarly to SP mode for campaigns
+   * Refactor the code which determines when actions can be undone, and when actions are sent to the server
  ### Lua API
+   * Overhaul of the API to use named tuples instead of `data[1]`, `data[2]`, etc
```

Comment:
This isn't an API thing. The API was changed to use named tuples in 1.18 or maybe even in 1.16. What my PR did was go through the code and update it all to use the named tuples.

I'm not even sure this is something that needs to be mentioned in the changelog, as it's essentially just updating older code.

Reply:
Okay, I skipped over the "obsolete" and misunderstood what the change was when converting commit messages into logs then. Same applies to the next two comments.

It does, however, appear to have changed the ai_helper's API, and thus be the final part of the API overhaul, even if the bulk of that work was done before.

Reply:
I don't remember, it's possible it might've changed the ai_helper API, yeah.